enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1120 @@
+| `x-forwarded-for`          | `haproxy.router.openshift.io/set-forwarded-headers` | Yes                                                               | Yes                                                   | Yes(using API spec.httpHeaders.forwardedHeaderPolicy) except delete action) | Yes(using route annotation)                          |
+| `x-ssl`                    | None                                                | Yes                                                               | Yes                                                   | None                                                                        | None                                                 |                    
+
+
+### Drawbacks
+
+Please refer to the `Risks and Mitigations` section to understand which headers are not allowed to be set/delete using this EP.
+
+## Design Details
+
+### Open Questions
+
+#### Will this enhancement cover setting or deleting request headers in addition to response headers?
+  Answer: Yes, we will be covering setting or deleting of request headers.
+  This is required for the following use-case.  
+  
+  As a cluster admin or route owner, I want to enable mutual TLS but replace the "x-ssl-client-der" request header
+  that OpenShift router adds with an "x-forwarded-client-cert" request header that my application checks for, 
+  so that I can migrate my applications that rely on mutual TLS and use the latter header name. 
+
+  The following example replaces "x-ssl-client-der" with "x-forwarded-client-cert":
+```yaml
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-SSL-Client-Cert
+        action:
+        type: Set
+        set:
+          value: "%{+Q}[ssl_c_der,base64]"
+      - name:   Content-Language
+        action:
+        type: Delete
+```
+
+#### Will this enhancement cover setting the load-balancing algorithm based on headers?
+  Answer: The HAProxy configuration provides a [load balancing algorithm](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-balance) which relies on the
+  check against the incoming request's header values.
+  
+  Even though this enhancement adds the header manipulation capability it
+  doesn't aim at providing (or enhancing) the load balancing API for the IngressController or Route. 
+  
+  Therefore, such a load balancing algorithm is out of scope for this enhancement and should be handled with a separate RFE and thereby it will be considered as a non-goal.
+
+#### Will this enhancement cover appending headers?
+  Answer: Yes and no.  This enhancement will not cover appending of
```

Comment:
Why "Yes and no"?
```suggestion
  Answer: No.  This enhancement will not cover appending of
```

Reply:
earlier this question had "delete".   So just forgot to delete yes.